core/src/main/java/org/frankframework/parameters/NumberParameter.java

```diff
@@ -110,27 +112,27 @@ public Object getValue(ParameterValueList alreadyResolvedParameters, Message mes
 	}
 
 	@Override
-	protected Number getValueAsType(Object request, boolean namespaceAware) throws ParameterException, IOException {
+	protected Number getValueAsType(@Nonnull Message request, boolean namespaceAware) throws ParameterException, IOException {
 		if(getType() == ParameterType.NUMBER) {
-			if (request instanceof Number number) {
+			if (request.asObject() instanceof Number number) {
 				return number;
 			}
 			log.debug("Parameter [{}] converting result [{}] to number decimalSeparator [{}] groupingSeparator [{}]", this::getName, ()->request, decimalFormatSymbols::getDecimalSeparator, decimalFormatSymbols::getGroupingSeparator);
 			DecimalFormat decimalFormat = new DecimalFormat();
 			decimalFormat.setDecimalFormatSymbols(decimalFormatSymbols);
-			try (Message requestMessage = Message.asMessage(request)) {
-				return decimalFormat.parse(requestMessage.asString());
+			try {
+				return decimalFormat.parse(request.asString());
 			} catch (ParseException e) {
 				throw new ParameterException(getName(), "Parameter [" + getName() + "] could not parse result [" + request + "] to number decimalSeparator [" + decimalFormatSymbols.getDecimalSeparator() + "] groupingSeparator [" + decimalFormatSymbols.getGroupingSeparator() + "]", e);
 			}
 		}
 		if(getType() == ParameterType.INTEGER) {
-			if (request instanceof Integer integer) {
+			if (request.asObject() instanceof Integer integer) {
```

Comment:
Ik vind er wel verdacht veel ~~message.asObject()~~ insluipen zo

Reply:
Klopt... We hebben in `Message` wel een method om te controleren of het request een instance is van XYZ, maar niet om dan (conditioneel) te unwrappen naar dat type. Een `Message.asTypeOrNull(Class requestedType)` of zoiets.